receiver/k8slogreceiver/config.go

```diff
@@ -0,0 +1,300 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package k8slogreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8slogreceiver"
+
+import (
+	"fmt"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.uber.org/multierr"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/consumerretry"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/k8sconfig"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator"
+)
+
+const (
+	ModeDaemonSetStdout = "daemonset-stdout"
+	ModeDaemonSetFile   = "daemonset-file"
+	ModeSidecar         = "sidecar"
+	StartAtBeginning    = "beginning"
+	StartAtEnd          = "end"
+)
+
+const (
+	DefaultFingerprintSize             = 1000
+	DefaultStartAt                     = StartAtEnd
+	DefaultEncoding                    = "utf-8"
+	DefaultMaxLogSize                  = 1024 * 1024
+	DefaultForceFlushPeriod            = 500 * time.Millisecond
+	DefaultPreserveLeadingWhitespaces  = true
+	DefaultPreserveTrailingWhitespaces = false
+	DefaultMaxReaders                  = 1000
+	DefaultIncludeFileName             = true
+	DefaultIncludeFilePath             = true
+	DefaultMode                        = ModeDaemonSetStdout
+	DefaultHostRoot                    = "/host_root"
+	DefaultNodeFromEnv                 = "K8S_NODE_NAME"
+)
+
+// Config is the configuration of a k8s input operator
+type Config struct {
```

Comment:
Are all of the configuration options required for the initial implementation? Can you remove them for now and add them along with the functionality using them going forward?

Reply:
I agree, starting small will make things easier to review and get started and we can always add them in the future.